ludwig/data/dataset/ray.py

```diff
@@ -352,9 +392,12 @@ def sync_read():
     def _create_async_reader(self, pipeline: DatasetPipeline):
         q = queue.Queue(maxsize=100)
         batch_size = self.batch_size
+        augment_batch = self._augment_batch_fn()
 
         def producer():
-            for batch in pipeline.iter_batches(prefetch_blocks=0, batch_size=batch_size, batch_format="pandas"):
+            for batch in pipeline.map_batches(augment_batch, batch_size=batch_size, batch_format="pandas").iter_batches(
```

Comment:
Nice!

Nit: if the augmentation pipeline is empty, we can skip calling `pipeline.map_batches`, which will save us an extra step in the pipeline. Something like:

```
def producer:
    if not augmentation_pipeline.empty():
        pipeline = pipeline.map_batches(augment_batch, batch_size=batch_size, batch_format="pandas")
```

Reply:
@tgaddair   made the change.  Take a look at the use of nonlocal.  I believe it is correct use.  Let me know.